src/components/Header/UserPreferences/Automation/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { useAutomation } from '../../../../@context/Automation/AutomationProvider'
+import Tooltip from '../../../@shared/atoms/Tooltip'
+import Details from './Details'
+import Transaction from '@images/transaction.svg'
+import Caret from '@images/caret.svg'
+import Lock from '@images/lock.svg'
+import classNames from 'classnames/bind'
+import stylesIndex from '../index.module.css'
+import styles from './index.module.css'
+
+const cx = classNames.bind(styles)
+
+export default function Automation(): ReactElement {
+  const {
+    autoWallet,
+    isAutomationEnabled,
+    hasValidEncryptedWallet,
+    balance,
+    nativeBalance
+  } = useAutomation()
+
+  const [hasError, setHasError] = useState<boolean>()
+  const [hasWarning, setHasWarning] = useState<boolean>(false)
+
+  useEffect(() => {
+    balance &&
+      setHasWarning(
+        Object.keys(balance)?.filter((token) => Number(balance[token]) <= 0)
+          .length > 0
+      )
+    setHasError(Number(nativeBalance?.balance) <= 0)
+  }, [balance, nativeBalance])
+
+  const wrapperClasses = cx({
+    automation: true,
+    enabled: isAutomationEnabled
+  })
+
+  const indicatorClasses = cx({
+    indicator: true,
+    enabled: isAutomationEnabled,
+    warning: hasWarning,
+    error: hasError
+  })
+
+  return (
+    <Tooltip
+      content={<Details isFunded={!hasError} />}
+      trigger="focus mouseenter click"
+      placement="bottom"
+      className={`${stylesIndex.preferences} ${wrapperClasses}`}
+    >
+      <div className={styles.wrapper}>
+        {!autoWallet && hasValidEncryptedWallet() && (
```

Comment:
nitpick: The hasValidEncryptedWallet function is used as a condition for rendering the Lock icon. It might be clearer to store its result in a state or a variable for better readability.

Reply:
Thanks, agreed. Having this be a callback also seems to be a leftover from previous iterations. I refactored this to be a state variable instead.